clang_build/target.py

```diff
@@ -198,6 +166,38 @@ def compile(self, process_pool, progress_disabled):
 
 
 class HeaderOnly(Target):
+    def __init__(self,
+            project_identifier,
+            name,
+            root_directory,
+            build_directory,
+            headers,
+            include_directories,
+            include_directories_public,
+            build_type,
+            clang,
+            clangpp,
+            options=None,
+            dependencies=None):
+        super().__init__(
+            project_identifier=project_identifier,
+            name=name,
+            root_directory=root_directory,
+            build_directory=build_directory,
+            headers=headers,
+            include_directories=[],
+            include_directories_public=include_directories_public+include_directories,
+            build_type=build_type,
+            clang=clang,
+            clangpp=clangpp,
+            options=options,
+            dependencies=dependencies)
+
+    def add_target_flags(self, target):
+        # Forward interface
+        self.compile_flags_interface += target.compile_flags_interface
+        self.link_flags_interface    += target.link_flags_interface
```

Comment:
Maybe I'm overlooking something, but shouldn't this function also forward public flags? The same goes for the static library `add_target_flags` function.

Also -- while I'm not strongly against this structure of using `add_target_flags` methods -- I think it was previously easier to read. The single if-statement in the `Target` class made it clear which targets forward what. Are there arguments for this design aside from aesthetics of the program structure?